lib/config/src/app/job.rs

```diff
@@ -0,0 +1,262 @@
+use std::{borrow::Cow, collections::HashMap, fmt::Display, str::FromStr};
+
+use serde::{de::Error, Deserialize, Serialize};
+
+use crate::package::PackageSource;
+
+use super::{pretty_duration::PrettyDuration, AppConfigCapabilityMemoryV1, AppVolume, HttpRequest};
+
+/// Job configuration.
+#[derive(
+    serde::Serialize, serde::Deserialize, schemars::JsonSchema, Clone, Debug, PartialEq, Eq,
+)]
+pub struct Job {
+    name: String,
+    trigger: JobTrigger,
+    #[serde(flatten)]
+    action: JobAction,
+}
+
+#[derive(
+    serde::Serialize, serde::Deserialize, schemars::JsonSchema, Clone, Debug, PartialEq, Eq,
+)]
+#[serde(rename_all = "lowercase")]
+pub enum JobAction {
+    Fetch(HttpRequest),
+    Execute(ExecutableJob),
+}
+
+#[derive(
+    serde::Serialize, serde::Deserialize, schemars::JsonSchema, Clone, Debug, PartialEq, Eq,
+)]
+pub enum CronType {
+    Daily,
+    Hourly,
+    Weekly,
+    CronExpression(String),
+}
+
+#[derive(Clone, Debug, PartialEq, Eq)]
+pub enum JobTrigger {
```

Comment:
Here we could just use `#[serde(rename_all = "kebab-case")]` so we don't need a custom De/Ser implementation.

We need that just for the CronType.

Reply:
if it was

```yaml
trigger:
  cron: "* * * * *"
```

then yeah, but currently it's just a string and we need to decide based on the string's format, so I don't think it's possible.